src/lib/components/autocompleteAddress/index.test.tsx

```diff
@@ -55,37 +55,41 @@ describe('AutocompleteAddress component', () => {
       }
     };
 
-    const screen = setup();
-    fireEvent.change(screen.getByTestId(inputTestId), {
+    const { getAllByTestId, getByTestId } = setup();
+
+    fireEvent.change(getByTestId(inputTestId), {
```

Comment:
Is there a reason to leave `fireEvent` here instead of using `userEvent`?

Reply:
How does `userEvent` differ from `fireEvent` technically?

Reply:
We should keep in mind, that `autocomplete` in this case adds a listener to the input completely devoid of the React life-cycle.

Reply:
As far as I know, userEvent is built on top of fireEvent but adds some "user like" goodies, like pointer up and down on click, checking the type of element being triggered and starting from that it has slightly different behaviour. 
You can check the implementation of here:
https://github.com/testing-library/user-event/blob/5feaa942f46bb37d96c2f2fbeb4b33e8beff75ad/src/click.js#L87-L103

Regarding the autocomplete, I agree it's something we should be aware of and, IMO, even take a second look to see if it's possible to update it to be part of the react cycle as you mentioned

Reply:
> We should keep in mind, that autocomplete in this case adds a listener to the input completely devoid of the React life-cycle.

True, I forgot about this custom implementation. Ok, let's stick with `userEvent`.
Adding up to what Diogo said regarding the differences, `userEvent` abstracts a little bit more the event functionalities and doesn't allow you to mess with internals. That gives you an environment more similar to what a user would experience.